change lti keys to number type

jxjj opened the pull request:
This:
- fixes type hinting for `resource_link_pk` and `lti13_resource_link_id`( these are `number | null`, not `string | null`)
- quiets a TS error by first checking for existence before doing the positive value check

Also:
- Updates tsconfig to exclude `vendor` and `node_modules` from type checking.

cmcfadden commented on resources/assets/js/views/ChimePage/ChimeManagement.vue:
```diff
@@ -126,23 +124,27 @@ import JoinPanel from "../../components/JoinPanel.vue";
 const store = useStore();
 
-// successful force sync
+const isCanvasChime = computed((): boolean => {
```
is there a reason we can't use your existing isCanvasChime selector here? Will that fail in some cases?

jxjj replied:
Good question! That should work! ty

jxjj commented:
- Switched to using existing `isCanvasChime` selector
- migrated selectors to TS
- added `ts-jest` to support jest tests with TS, migrated selector tests to TS as well
